### PR TITLE
Renames console to Matomo

### LINKED ISSUE #12924: Console is named Piwik instead of Matomo plus matomo-domain option
Currently, the name of the console is `Piwik` instead of `Matomo` and we should also add support for `matomo-domain` which should take precedence over `piwik-domain`. Eventually, also need to adjust docs to say `--matomo-domain=` instead of `--piwik-domain=...`

### PR DESCRIPTION
**Pull request by sgiehl**
and changes --piwik-domain option to --matomo-domain

fixes #12924

### CODE REVIEW

File: core/Console.php [hunk 1]
```diff
@@ -31,12 +31,12 @@ public function __construct(Environment $environment = null)
...
 
-        parent::__construct('Piwik', Version::VERSION);
+        parent::__construct('Matomo', Version::VERSION);
 
         $this->environment = $environment;
 
-        $option = new InputOption('piwik-domain',
-            null,
+        $option = new InputOption('matomo-domain',
+            'piwik-domain', // @todo  Remove this alias in Matomo 4.0
```
Comment (sgiehl) [inline core/Console.php]: Is adding `piwik-domain` as an alias of `matomo-domain` fine for BC?

**Comment by diosmosis** [comment]
Would the eventual removal of `--piwik-domain` be put in a list somewhere? Just to protect against forgetting.

**Review by diosmosis — APPROVED** [review]
Left a comment, but looks good.

**Comment by sgiehl** [comment]
> Would the eventual removal of --piwik-domain be put in a list somewhere? Just to protect against forgetting.

@diosmosis  updated the list in https://github.com/matomo-org/matomo/issues/12420

**Pull request merged**